### PR TITLE
tests: add tests to RoleFilter component

### PR DESCRIPTION
**Pull request by Isaac-alencar**
Closes: [issue-52](https://github.com/comarev/comarev-dashboard/issues/52)

References: <issue-link>/<pr-link>

#### What?

- Add tests for RoleFilter component
#### Why?

- To ensure the component is returning null or an given element depending if user has or not proper roles

#### How to test it?

```bash
# to run all tests suits
yarn test

# press 'a' to run all tests and see tests for RoleComponents passing
```

### CODE REVIEW

File: src/components/role-filter/__tests__/role-filter.component.test.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+const mockElement = () => <div>some_content_for_element</div>;
+
+describe('RoleComponent', () => {
+  describe('when user doesn\'t have proper roles', () => {
+    beforeEach(() => {
+      useSelector.mockImplementation(() => userData);
+    });
+
+    it('should return null instead of given children element', () => {
+
```
Comment (geeksilva97) [inline src/components/role-filter/__tests__/role-filter.component.test.js:19]:
```suggestion
```

File: src/components/role-filter/__tests__/role-filter.component.test.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
+import { useSelector } from 'react-redux';
+import { userData } from '../../../test/mocks/user';
+import RoleFilter from '../role-filter.component';
+
+jest.mock('react-redux', () => ({
```
Comment (belgamo) [inline src/components/role-filter/__tests__/role-filter.component.test.js:5]: You don't need to mock this. Use our `wrapper` instead.

File: src/components/role-filter/__tests__/role-filter.component.test.js (hunk 1 above)
Comment (belgamo) [inline src/components/role-filter/__tests__/role-filter.component.test.js:19]: Remove empty lines :nail_care:

**Review by belgamo — CHANGES_REQUESTED** [review]
Overall looks great! As you're here adding tests, I'll take advantage and ask you to abstract the logic into a custom hook, something like `useRoleFilter` return the `hasPermission`. Then you just have to call this hook in the component and we can reuse this hook over our app.

**Comment by Isaac-alencar** [comment]
Not quite sure if what I've done on `useRoleFilter` tests is the more appropriated approach. Following the documentation of `@testing-library/react-hooks` in [this section](https://react-hooks-testing-library.com/usage/advanced-hooks), to render hooks that needs information from some provider(_e.g redux_), we need to use a wrapper, and this wrapper needs to be simple as:
```jsx
const wrapper = ({ children }) => <CounterStepProvider step={2}>{children}</CounterStepProvider>
```
But our wrapper returns a rendered element and didn't work as expected. If you guys have a tip to improve my tests I will be thankful

**Comment by belgamo** [comment]
> Not quite sure if what I've done on `useRoleFilter` tests is the more appropriated approach. Following the documentation of `@testing-library/react-hooks` in [this section](https://react-hooks-testing-library.com/usage/advanced-hooks), to render hooks that needs information from some provider(_e.g redux_), we need to use a wrapper, and this wrapper needs to be simple as:
> 
> ```js
> const wrapper = ({ children }) => <CounterStepProvider step={2}>{children}</CounterStepProvider>
> ```
> 
> But our wrapper returns a rendered element and didn't work as expected. If you guys have a tip to improve my tests I will be thankful

I've pushed a commit refactoring the `wrapper` component to support hooks.

**Pull request merged**